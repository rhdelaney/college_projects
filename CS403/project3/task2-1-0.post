
(define (a + x y)
    (+ (x y) (b (c +)))
    )
(define (b @)
    (display (quote (+ + +)))
    (* (g - @) (c + @ @) (d (car @)))
    )
(define (c op quote)
    (op quote quote)
    )
(define (d items)
    (g - items)
    )

(include "pretty.lib")
(compile a)
(pretty a)
(pretty b)
(pretty c)
(pretty d)

