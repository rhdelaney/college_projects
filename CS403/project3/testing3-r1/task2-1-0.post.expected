(define (a + x y)
    (+ (x y) (<function b(@)> (<function c(op quote)> +)))
    )
(define (b @)
    (<builtin display(item)> (<builtin quote($item)> (+ + +)))
    (<builtin *(@)> (g <builtin -(@)> @) (<function c(op quote)> <builtin +(@)> @ @) (<function d(items)> (<builtin car(items)> @)))
    )
(define (c op quote)
    (op quote quote)
    )
(define (d items)
    (g <builtin -(@)> items)
    )
